### PR TITLE
Enable ansible-lint

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/1663
Enable ansible-lint

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by gravesm — COMMENTED** [review]
I guess I would probably add ansible-lint to the linters environment like we do with black. This would simplify the zuul config some.

**Comment by alinabuzachis** [comment]
> I guess I would probably add ansible-lint to the linters environment like we do with black. This would simplify the zuul config some.

Tried to do that. So, we do not need anymore this one https://github.com/ansible/ansible-zuul-jobs/pull/1663, right?

File: tox.ini [hunk 1]
```diff
@@ -14,16 +14,23 @@ deps =
...
   yamllint
   flake8
   {[testenv:black]deps}
+  {[testenv:ansible-lint]deps}
 
 commands =
   black -v --check --diff {toxinidir}/plugins {toxinidir}/tests
   yamllint {toxinidir}
   flake8 {toxinidir}
+  ansible-lint --profile production --format pep8 --nocolor --strict {toxinidir}/plugins {toxinidir}/roles
```
Comment (gravesm) [inline tox.ini:33]: You can also use the substitution here if you want, like `{[testenv:ansible-lint]commands}`

Comment (alinabuzachis) [reply tox.ini]: Right, thank you! Cleaner this way!

**Comment by gravesm** [comment]
Do we need the GHA anymore? Isn't that just a subset of what this PR adds with zuul?

**Comment by alinabuzachis** [comment]
> Do we need the GHA anymore? Isn't that just a subset of what this PR adds with zuul?

I do not think we need it. Just removed.

**Comment by gravesm** [comment]
Given that this PR now makes a bunch of changes to the content, not just test config, we should add a changelog fragment.

**Review by gravesm — APPROVED** [review]

**Review by abikouo — APPROVED** [review]

**Review by jillr — APPROVED** [review]

**Pull request merged**